### PR TITLE
Introduce PHIVE_HOME environment variable solves #286

### PR DESCRIPTION
**Pull request by dnaber-de**
PR for #286 

If I missed anything please let me know

### CODE REVIEW

File: src/shared/config/Config.php
Comment (theseer) [inline src/shared/config/Config.php:41]: Yoda should stay with Star Wars ;)

File: src/shared/config/Config.php
Comment (theseer) [inline src/shared/config/Config.php:42]: This is a 2nd call to `getenv`. Please don't do that.

File: src/shared/config/Config.php
Comment (theseer) [reply src/shared/config/Config.php:41]: The use of `getenv` directly relies on global state. 

Please move that into the `Environment` class.

File: tests/unit/shared/config/ConfigTest.php
Comment (theseer) [inline tests/unit/shared/config/ConfigTest.php:15]: If you wouldn't have the implementation rely on global state, this wouldn't be required here.

File: tests/unit/shared/config/ConfigTest.php
Comment (theseer) [inline tests/unit/shared/config/ConfigTest.php:45]: No need to actually change the environment (global state) here if properly abstracted via `Environment`.

File: tests/unit/shared/config/ConfigTest.php
Comment (theseer) [inline tests/unit/shared/config/ConfigTest.php:59]: See above.

**Review by theseer — CHANGES_REQUESTED** [review]
Thanks for this PR.

Before I can merge this, please adjust the code as commented.

File: src/shared/config/Config.php
Comment (dnaber-de) [reply src/shared/config/Config.php:41]: To be clear `PharIo\Phive\Environment::getVariable()` should be used as accessor to environment variables?

File: tests/unit/shared/config/ConfigTest.php
Comment (dnaber-de) [reply tests/unit/shared/config/ConfigTest.php:15]: So within unit tests there shouldn't be any testing with real environment variables?

File: src/shared/config/Config.php
Comment (theseer) [reply src/shared/config/Config.php:41]: No, sorry for not being clear and good that you ask :+1: 

If I'm not mistaken, the `$_SERVER` array should already contain all environment variables. Thus, `Environment` should only require additional methods, e.g. something like `Environment::getPhiveHome` and `Environment::hasPhiveHomeVariable` (actual names subject to discussion ;) )

File: tests/unit/shared/config/ConfigTest.php
Comment (theseer) [reply tests/unit/shared/config/ConfigTest.php:15]: Yes and no:

- When testing `Config`, a clear __no__ as it should request everything via `Environment` - which can get mocked

- When testing `Environment`, for a unit test you can instantiate `Environment` with a controlled data set (e.g. pass in a custom array to its constructor). This will allow testing of all aspects.

- When testing the factory method `fromGlobalState`, we would need actual environment variables. As that piece of code interacts with and depends on global state, this is - formally speaking - and in my opinion not a Unit but an integration test.

File: src/shared/config/Config.php
Comment (theseer) [reply src/shared/config/Config.php:41]: Maybe to make it even more clear: The `getVariable` method is generic and only exists in case we need some component to query for a variable we don't know about up front and thus cannot write an explicit getter for.

Comment (dnaber-de) [reply src/shared/config/Config.php:41]: > I'm not mistaken, the $_SERVER array should already contain all environment variables.

That is what I was not sure about but you're right:
```
$ PHIVE_HOME="/whatever" php -r "var_dump(\$_SERVER);" | grep -A1 PHIVE
  'PHIVE_HOME' =>
  string(9) "/whatever"
```

Now everything is clear, I'll update the PR

File: src/shared/environment/Environment.php
Comment (dnaber-de) [inline src/shared/environment/Environment.php:136]: php-cs-fixer did that. Didn't want to spend another commit just on that one line.

File: src/shared/environment/Environment.php
Comment (dnaber-de) [inline src/shared/environment/Environment.php:48]: I decided to place these method next to `hasHomeDirectory()` as they are somehow related.

**Comment by dnaber-de** [comment]
@theseer Ready for second review

**Comment by theseer** [comment]
Looks all good!

**Pull request merged**
